src/validation.cpp

```diff
@@ -895,6 +895,20 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
         // This is done last to help prevent CPU exhaustion denial-of-service attacks.
         PrecomputedTransactionData txdata(tx);
         if (!CheckInputs(tx, state, view, true, scriptVerifyFlags, true, false, txdata)) {
+            // Assume that this is due to a policy violation, rather than
+            // re-checking with the consensus flags. Separating policy rules
+            // from consensus rules is tricky, because we want to be able to
+            // support old peers who may not be aware of recent consensus
+            // changes; returning NOT_STANDARD here is maximally permissive.
+            // Anyway re-checking the scripts with a different set of flags
+            // wastes CPU as well (particularly if an attacker is constructing
+            // transactions designed to waste our CPU).
+            state.Invalid(ValidationInvalidReason::TX_NOT_STANDARD, false, REJECT_NONSTANDARD, state.GetRejectReason(), state.GetDebugMessage());
```

Comment:
nit: maybe include in the comment why we have to call this function anyway (that CheckInputs always returns CONSENSUS so we have to overwrite it).

Reply:
Done